text/0715-construct-upgrade-support.md

```diff
@@ -0,0 +1,514 @@
+production environments.
+
+### How it works
+
+The CDK team develops a construct upgrade framework designed to simplify adoption while minimizing
+risks. This approach combines guided documentation, automated validations, and deployment safeguards
+to ensure a controlled transition.
+
+In a high-level overview, for each supported migration, the construct upgrade framework provides:
+
+- A comprehensive, step-by-step migration guide enabling users to transition from legacy constructs/modules
+to their modern equivalents.
+- A CDK CLI command `cdk construct-upgrade` to validate code changes and execute deployments safely.
+
+Note that each migration guide follows a standardized structure that includes dependency changes, behavioral
+analysis, risk assessments, and rollback instructions. This ensures consistency, clarity, and safety across
+all migrations. See [Appendix A](#appendix-a-migration-guide) for a detailed breakdown of the proposed migration
+guide format.
+
+With the initial release of the CDK construct upgrade tool, developers use the migration guide to
+update their CDK code, replacing legacy constructs (e.g., Table) with modern equivalents (e.g., TableV2).
+Developers have the flexibility to implement changes either manually or with the assistance of AI-powered
+code transformation tools (e.g., LLMs).
+
+After updating the code to use the new construct, the CDK CLI command `cdk construct-upgrade` validates the
+existing CDK stack and environment, ensuring they are ready for migration. The tool flags any warnings or
+blockers, such as CloudFormation drift, and cross-compares the old and new CloudFormation templates.
+It analyzes the CloudFormation change set and validates any library-specific configurations.
+
+Users can then choose to run the `cdk construct-upgrade` CLI command with:
+
+- The `--dry-run` option to perform validation only and generate a validation JSON file. Users can re-run
+the `cdk construct-upgrade` CLI command, which will automatically check the `construct-upgrade.context.json`
+file to determine whether validations are complete before proceeding to execution and deployment.
+- If the `--dry-run` option is not specified, the CLI command will perform validation and automatically
+proceed with execution and deployment if the validations are successful.
+
+Alternatively, users can use `cdk deploy --construct-upgrade` with construct upgrade option to validate
+their construct upgrade changes before deployment starts.
+
+In the initial release of the CDK construct upgrade tool, the CLI supports migration of VPC-related constructs
+from `aws-cdk-lib/aws-ec2` to `@aws-cdk/aws-ec2-alpha` and `Table` to `TableV2` in `aws-cdk-lib/aws-dynamodb`.
+While the tool is designed to handle a wide range of construct migrations, the migration guides for these two
+upgrades will be the first available, with additional guides planned for future releases.
```

Comment:
Whatever mechanism we employ for this should be open and pluggable, right? Worth calling out whether it is or isn't.